src/Driver/RemoteClient.php

```diff
@@ -313,10 +313,12 @@ private function write(string $data, bool $close = false): Future
 
         if ($close) {
             $this->status |= self::CLOSED_WR;
-            return $this->socket->end($data);
+            $this->socket->end($data);
```

Comment:
`end` does no longer accept optional data, so this needs to be converted into separate `write` and `end` calls.